k8s/k8s.go

```diff
@@ -109,7 +100,13 @@ func CmdAddK8s(args *skel.CmdArgs, conf utils.NetConf, hostname string, calicoCl
 		endpoint.Metadata.Hostname = hostname
 		endpoint.Metadata.OrchestratorID = orchestratorID
 		endpoint.Metadata.WorkloadID = workloadID
-		endpoint.Spec.Profiles = []string{profileID}
+		// Set the profileID according to whether Kubernetes policy is required. If it's not, then just use the network
+		// name (which is the normal behavior) otherwise use one based on the Kubernetes pod namespace.
+		if conf.Policy.PolicyType == "" {
```

Comment:
I think this if statement should be the other way around (so that it matches the label check below).

We only want to set the per-namespace profile if we're using `PolicyType == "k8s"`.